src/dotty/tools/dotc/transform/VCInlineEqAndField.scala

```diff
@@ -7,20 +7,18 @@ import Contexts._, Trees._, StdNames._, Symbols._
 import DenotTransformers._, TreeTransforms._, Phases.Phase
 import ExtensionMethods._, TreeExtractors._, ValueClasses._
 
-/** This phase inlines calls to methods and fields of value classes.
+/** This phase inlines calls to == and to the field of a value class
```

Comment:
I would prefer to treat all methods equally, not special-casing `==`, that could also be specialized & etc.